libqtile/bar.py

```diff
@@ -273,24 +274,30 @@ def _configure(self, qtile, screen):
             self.window.opacity = self.opacity
             self.window.unhide()
 
-            self.drawer = self.window.create_drawer(width, height)
-            self.drawer.clear(self.background)
-
             self.window.process_window_expose = self.process_window_expose
+            self.window.process_window_expose = self.draw
             self.window.process_button_click = self.process_button_click
             self.window.process_button_release = self.process_button_release
             self.window.process_pointer_enter = self.process_pointer_enter
             self.window.process_pointer_leave = self.process_pointer_leave
             self.window.process_pointer_motion = self.process_pointer_motion
             self.window.process_key_press = self.process_key_press
 
+        # We create a new drawer even if there's already a window to ensure the
+        # drawer is the right size.
+        self.drawer = self.window.create_drawer(width, height)
+        self.drawer.clear(self.background)
```

Comment:
Can we not just resize the existing `Drawer`?

Reply:
We may want a `Drawer.resize()` method as we'd need to recreate a new `XCBSurface` or `ImageSurface` with the new size. Seemed quicker just to make a new `Drawer`.

Reply:
It looks like the x11 `Drawer` supports resizing but using setters for `width` and `height` properties. AFAICT this isn't supported by the Wayland implementation though. But yeah if it's easier to just create a new one then go for that :)